tests/CTestCustom.cmake

```diff
@@ -162,10 +148,12 @@ if ((NOT "@FORCE_RUN_ALL_TESTS@" STREQUAL "ON") AND (NOT "@USE_SANITIZER@" STREQ
     )
   endif()
 
-  if("@USE_SANITIZER@" MATCHES "Undefined" AND "@CMAKE_CXX_COMPILER_ID@" STREQUAL "Clang")
+  if("@USE_SANITIZER@" MATCHES "Undefined" AND "@CMAKE_CXX_COMPILER_ID@" STREQUAL "Clang" AND "@CMAKE_CXX_COMPILER_VERSION@" VERSION_LESS_EQUAL "20.0.0")
```

Comment:
It's either 20 or 21 the first version for which these test pass but I don't have an easy way to test with 20.